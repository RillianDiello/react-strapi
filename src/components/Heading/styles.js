import styled, { css } from 'styled-components';

const titleSizes = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.huge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1.withConfig({
  shouldForwardProp: (prop) =>
    !['colorDark', 'size', 'uppercase'].includes(prop),
})`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSizes[size] ? titleSizes[size](theme) : titleSizes.medium(theme)};
    ${titleCase(uppercase)};
  `}
`;
