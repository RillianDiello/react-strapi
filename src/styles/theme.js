export const theme = {
  colors: {
    mainBg: 'gray',
    secondaryBg: 'white',
  },
  fonts: {},
  spacings: {},
};
