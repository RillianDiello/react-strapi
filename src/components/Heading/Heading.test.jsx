import 'jest-styled-components';
import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/theme';
import { Heading } from '.';

describe('<Heading />', () => {
  it('should render with default values', () => {
    render(
      <ThemeProvider theme={theme}>
        <Heading colorDark>text</Heading>
      </ThemeProvider>,
    );
    const heading = screen.getByRole('heading', { name: /text/i });
    expect(heading).toHaveStyle({
      color: theme.colors.primaryColor,
    });
  });
  it('should render with dark color', () => {
    render(
      <ThemeProvider theme={theme}>
        <Heading colorDark={false}>text</Heading>
      </ThemeProvider>,
    );
    const heading = screen.getByRole('heading', { name: /text/i });
    expect(heading).toHaveStyle({
      color: theme.colors.white,
    });
  });
  it('should render with uppercase letters', () => {
    render(
      <ThemeProvider theme={theme}>
        <Heading uppercase={true}>text</Heading>
      </ThemeProvider>,
    );
    const heading = screen.getByRole('heading', { name: /text/i });
    expect(heading).toHaveStyle({
      'text-transform': 'uppercase',
    });
  });
  it('should render correact heading element', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Heading as="h6">text</Heading>
      </ThemeProvider>,
    );
    const h6 = container.querySelector('h6');
    expect(h6.tagName.toLowerCase()).toBe('h6');
  });
});
