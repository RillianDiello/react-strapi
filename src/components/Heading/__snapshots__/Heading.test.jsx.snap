// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render with default values 1`] = `
<h1
  class="sc-blHHSb bepzXa"
>
  TÃ­tulo
</h1>
`;
